### PR TITLE
Add asAB option to getPhotometry to return AB not nJy

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: python/pfs/datamodel/pfsConfig.py [hunk 1]
```diff
@@ -436,7 +436,8 @@ def getVariant(self):
         """Return the (variantNum, basePfsDesign) pair, or (0, 0) if we are not a variant """
         return self.variant, self.designId0
 
-    def getPhotometry(self, filterName, psfFlux=False, fiberFlux=False, totalFlux=False, getError=False):
+    def getPhotometry(self, filterName, psfFlux=False, fiberFlux=False, totalFlux=False, getError=False,
+                      asAB=False):
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:440]: `asAbMag`?

File: python/pfs/datamodel/pfsConfig.py [hunk 2]
```diff
@@ -481,19 +484,46 @@ def getPhotometry(self, filterName, psfFlux=False, fiberFlux=False, totalFlux=Fa
...
+        def nJyToAB(nJy):
+            return 8.9 - 2.5*np.log10(1e-9*nJy)
 
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=RuntimeWarning)  # All-NaN slice encountered
 
             flux = np.nanmean(np.where(myFilterData, flux, np.NaN), axis=1)
 
+            if asAB:
+                AB = nJyToAB(flux)
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:511]: Variables must not be named starting with a capital; that is reserved for class names.

File: python/pfs/datamodel/pfsConfig.py [hunk 3]
```diff
@@ -481,19 +484,46 @@ def getPhotometry(self, filterName, psfFlux=False, fiberFlux=False, totalFlux=Fa
...
+
             if getError:
                 fluxErr = np.nanmean(np.where(myFilterData, fluxErr, np.NaN), axis=1)
-                return flux, fluxErr
+
+                if asAB:
+                    if fluxErr < flux:
+                        ABErr = np.mean([nJyToAB(flux + fluxErr) - AB, AB - nJyToAB(flux - fluxErr)])
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:518]: Ditto.

**Review by PaulPrice — APPROVED** [review]
Minor comments, otherwise good.

File: python/pfs/datamodel/pfsConfig.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply python/pfs/datamodel/pfsConfig.py:511]: Flake8 is happy, I suspect that all caps is OK for a variable.  But I'll change it, if only to be consistent with `asAbMag`

**Pull request merged**